cpp/sopt/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ set(headers
   maths.h proximal.h relative_variation.h sdmm.h
   wavelets.h conjugate_gradient.h l1_proximal.h logging.enabled.h padmm.h proximal_expression.h
   reweighted.h types.h wrapper.h exception.h linear_transform.h logging.h positive_quadrant.h
-  real_type.h sampling.h power_method.h ${PROJECT_BINARY_DIR}/include/sopt/config.h
+  real_type.h sampling.h power_method.h objective_functions.h ${PROJECT_BINARY_DIR}/include/sopt/config.h
```

Comment:
@Luke-Pratley - this was making an example on purify to fail. Could you check that this indeed needs to go here?